storagenode/contact/service.go

```diff
@@ -34,13 +35,17 @@ type Service struct {
 
 	mu   sync.Mutex
 	self *overlay.NodeDossier
+
+	gotSelfData bool
```

Comment:
See https://github.com/storj/storj/blob/master/internal/sync2/fence.go#L13

Reply:
Aha! I was sure we had something like that but didn't look in the right place.

Reply:
Wait, nope, I don't think I want to use Fence; it's not cancelable via Context.

Reply:
PS: It's possible to change sync.Fence code :P

Reply:
True. It would be a bit of a fundamental change, though, and with something like that which is clearly written carefully for performance and correctness, I worry that I'd mess up some important semantics that something else is relying on.